docs/proposals/promote-dependencies-automatically/README.md

```diff
@@ -0,0 +1,196 @@
+---
+title: Promote Dependencies Automatically
+authors:
+- "@zhy76" # Authors' github accounts here.
+reviewers:
+- "@jwcesign"
+- "@chaunceyjiang"
+approvers:
+- TBD
+
+creation-date: 2023-10-06
+
+---
+
+# Promote Dependencies Automatically
+
+## Summary
+
+One resource could only be okay when the dependencies are ready, such as Deployment depending on Configmap. So, If we want to promote the resource and deploy it into multiple clusters, we need to promote the resource and all the dependencies, it would be great if this process could be done automatically.
+
+Therefore, we need to provide an automatic promotion mechanism for dependent resources in karmadactl. When you promote a resource, it is necessary to automatically promote all the resources on which that resource depends. For example, promote the Secret that Deployment depends on.
+
+Typically, users don't create Pods directly. Instead, users create them using workload resources such as Deployment, StatefulSet, or Job. Take Deployment as an example. As shown in the following figure, the dependent resources of Deployment include ConfigMaps, Secrets, PVC, and Services.
+
+And, in addition to the native resources in kubernetes, the resources on which the user's CRD/CR resources depend need to support automatic promotion.
+
+![dependencies](statics/dependenciesOfDeployment.drawio.svg)
```

Comment:
What's the difference between the different colored dependency markers in this diagram, I see them marked in three different colors.

Reply:
I borrowed this from the diagram here 😸 : https://github.com/karmada-io/karmada/blob/master/docs/proposals/dependencies-automatically-propagation/dependenciesOfDeployment.png
Maybe I need to make it the same color？

Reply:
Or maybe being able to explain what the different colors mean would be a little better, ignore it for now.